src/blocks/select/Select.tsx

```diff
@@ -0,0 +1,240 @@
+};
+
+export type SelectProps = {
+  options: SelectOption[];
+  onSelect?: (value: string) => void;
+  selectedValue?: string;
+  css?: FlattenSimpleInterpolation;
+  description?: string;
+  disabled?: boolean;
+  error?: boolean;
+  errorMessage?: string;
+  label?: string;
+  placeholder?: string;
+  required?: boolean;
+  success?: boolean;
+};
+
+const Container = styled.div<{ css?: FlattenSimpleInterpolation }>`
+  align-items: flex-start;
+  display: flex;
+  flex-direction: column;
+  flex: 1 0 0;
+  gap: var(--spacing-xxs, 8px);
+
+  /* Custom CSS applied via styled component css prop */
+  ${(props) => props.css || ''};
+`;
+
+const StyledBox = styled.div<{
+  error?: boolean;
+  success?: boolean;
+  disabled?: boolean;
+}>`
+  ${({ theme, success, error, disabled }) => {
+    const colors = theme?.blocksTheme?.colors;
+    const defaultState = error ? 'danger' : success ? 'success' : disabled ? 'disabled' : 'default';
+    const focusState = error ? 'danger' : success ? 'success' : 'focus';
+
+    return css`
+      display: flex;
+      align-self: stretch;
+      align-items: center;
+      justify-content: space-between;
+      border-radius: var(--radius-xs, 12px);
+      border: 1.5px solid
+        var(--components-inputs-stroke-${defaultState}, ${colors[`components-inputs-stroke-${defaultState}`]});
+      background: var(
+        --components-inputs-background-${defaultState},
+        ${colors[`components-inputs-background-${defaultState}`]}
+      );
+      padding: var(--spacing-xs, 12px);
+      &:hover {
+        border: 1.5px solid var(--components-inputs-stroke-hover, #c4cbd5);
+      }
+
+      &:focus-within {
+        border: 1.5px solid
+          var(--components-inputs-stroke-${focusState}, ${colors[`components-inputs-stroke-${focusState}`]});
+        outline: none;
+      }
+
+      [data-reach-combobox-input] {
+        background-color: transparent;
+        border: none;
+        color: var(--components-inputs-text-default, ${colors['components-inputs-text-default']});
+
+        display: flex;
+
+        font-family: var(--font-family);
+        font-size: ${textVariants['bs-regular'].fontSize};
+        font-style: ${textVariants['bs-regular'].fontStyle};
+        font-weight: ${textVariants['bs-regular'].fontWeight};
+        line-height: ${textVariants['bs-regular'].lineHeight};
+
+        gap: var(--spacing-none, 0px);
+
+        &:focus {
+          outline: none;
+        }
+        &:hover {
+          outline: none;
+        }
+        &:disabled {
+          background-color: transparent;
+          cursor: not-allowed;
+          color: var(--components-inputs-text-disabled, ${colors['components-inputs-text-disabled']});
+        }
+
+        ::placeholder {
+          color: var(--components-inputs-text-placeholder, ${colors['components-inputs-text-placeholder']});
+        }
+      }
+    `;
+  }}
+`;
+
+const StyledPopover = styled(ComboboxPopover)<{ popoverWidth: number }>`
```

Comment:
When popover is opened the scroll should be blocked similar to Radix UI select.

If this takes time. Create a separate issue and pick it up afterwards.